### PR TITLE
Accept baselines for #8288

### LINKED ISSUE #8288: [Transforms] Namespace with same name as default export uses exported reference instead of local
**Expected**:

```js
//// [defaultExportsCannotMerge04.ts]

export default function Foo() {
}

namespace Foo {
    export var x;
}

interface Foo {
}

export interface Foo {
}

//// [defaultExportsCannotMerge04.js]
"use strict";
function Foo() {
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = Foo;
var Foo;
(function (Foo) {
})(Foo || (Foo = {}));
```

**Actual**:

```js
//// [defaultExportsCannotMerge04.ts]

export default function Foo() {
}

namespace Foo {
    export var x;
}

interface Foo {
}

export interface Foo {
}

//// [defaultExportsCannotMerge04.js]
"use strict";
function Foo() {
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = exports.Foo;
var Foo;
(function (Foo) {
})(exports.Foo || (exports.Foo = {}));
```

Failed tests:
*  defaultExportsCannotMerge04.ts

### PR DESCRIPTION
**Pull request by mhegazy**
Closes #8288

### CODE REVIEW

File: tests/baselines/reference/defaultExportsCannotMerge04.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ export interface Foo {
 function Foo() {
 }
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.default = Foo;
+exports.default = exports.Foo;
```
Comment (rbuckton) [inline tests/baselines/reference/defaultExportsCannotMerge04.js]: :-1: This is incorrect as `Foo` is not an exported name.

File: tests/baselines/reference/defaultExportsCannotMerge04.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ export interface Foo {
 function Foo() {
 }
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.default = Foo;
+exports.default = exports.Foo;
 var Foo;
 (function (Foo) {
-})(Foo || (Foo = {}));
+})(exports.Foo || (exports.Foo = {}));
```
Comment (rbuckton) [inline tests/baselines/reference/defaultExportsCannotMerge04.js]: Also incorrect as `Foo` is not exported.

Comment (mhegazy) [inline tests/baselines/reference/defaultExportsCannotMerge04.js]: the issue is we do not expect something to be exported with its name and as default. it is either this or the other.

Comment (mhegazy) [inline tests/baselines/reference/defaultExportsCannotMerge04.js]: we do show the error for it.

**Pull request closed without merging**